Sources/Testing/RepeatTestCase.swift

```diff
@@ -45,7 +45,10 @@ open class RetryTestCase: ProcedureKitTestCase {
         let info = RetryTestCaseInfo()
         return AnyIterator {
             let procedure = TestProcedure()
-            procedure.add(condition: BlockCondition { info.numberOfFailures == failureThreshold })
+            procedure.add(condition: BlockCondition {
+                guard info.numberOfFailures == failureThreshold else { throw ProcedureKitError.conditionFailed() }
+                return true
+            })
```

Comment:
This change is a result of the changed behaviour of `BlockCondition` - not 100% sure on this breaking change.